vk_music_downloader_decorator.py

```diff
@@ -0,0 +1,132 @@
+        if len(args) == 3:
+            print()
+            print("The list of your tracks:")
+        fn(*args)
+        if len(args) == 4:
+            print("Download complete!")
+        print()
+        print("Have a nice day!")
+    return wrapper
+
+# Print the list of tracks sorted by artist / adding date
+@check_if_artist_exist
+def show_tracks(artist, tracks, sort):
+    artist_title = []
+    print()
+    if artist is None:
+        for track in tracks['items']:
+            artist_title.append(track['artist'] + " - " + track['title'])
+    else:
+        artist = artist[0]
+        for track in tracks['items']:
+            if track['artist'] == artist:
+                artist_title.append(track['artist'] + " - " + track['title'])
+    if sort == True:
+        artist_title.sort()
+    print('\n'.join(artist_title))
+
+
+# Download the specified tracks / all
+@check_if_artist_exist
+def download_tracks(artist, tracks, title, path):
+    if not os.path.exists(path):
+        os.makedirs(path)
+    if artist is None and title is None:
+        confirm = input("It seems you want to DOWNLOAD ALL THE TRACKS! Are you SURE? (Y / N) ")
+        if confirm.lower() == 'y':
+            print()
+            for track in tracks['items']:
+                url = track['url']
+                print('Downloading: {} - {}'.format(track['artist'], track['title']))
+                urlretrieve(url, path + '/' + track['artist'] + ' - ' + track['title'] + ".mp3")
+        if confirm.lower() == 'n':
+            sys.exit("Please, specify the arguments and try again!")
+        else:
+            sys.exit("It seems like you didn't choose YES.\nPlease, specify the arguments and try again!")
+    elif artist is not None:
+        artist = artist[0]
+        if title is None:
+            print()
+            for track in tracks['items']:
+                if track['artist'] == artist:
+                    url = track['url']
+                    print('Downloading: {} - {}'.format(track['artist'], track['title']))
+                    urlretrieve(url, path + '/' + track['artist'] + ' - ' + track['title'] + ".mp3")
+        else:
+            title = title[0]
+            print()
+            for track in tracks['items']:
+                if track['artist'] == artist and track['title'] == title:
+                    url = track['url']
+                    print('Downloading: {} - {}'.format(track['artist'], track['title']))
+                    urlretrieve(url, path + '/' + track['artist'] + ' - ' + track['title'] + ".mp3")
+    elif artist is None:
+        title = title[0]
+        print()
+        for track in tracks['items']:
+            if track['title'] == title:
+                url = track['url']
+                print('Downloading: {} - {}'.format(track['artist'], track['title']))
+                urlretrieve(url, path + '/' + track['artist'] + ' - ' + track['title'] + ".mp3")
```

Comment:
yep) it's was obvious in the v.1.0.0 yet. but i am a little bit ̶l̶a̶z̶y̶ busy) and decided to fix that in the next iteration)